src/bin/tikv-server.rs

```diff
@@ -812,6 +817,16 @@ fn get_store_labels(matches: &ArgMatches, config: &toml::Value) -> HashMap<Strin
         .unwrap_or_else(|err| exit_with_err(format!("{:?}", err)))
 }
 
+fn check_compression_available(compression: DBCompressionType) -> bool {
+    supported_compression().contains(&compression)
+}
+
+fn get_fastest_supported_compression_type() -> DBCompressionType {
```

Comment:
Why not move it to util module and fetch it when necessary (like generating snap file)? So you don't have to change so many places and `SnapManager` and other structs don't have to accept another parameter.